src/OrbitQt/TargetLabel.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_TARGET_LABEL_H_
+#define ORBIT_QT_TARGET_LABEL_H_
+
+#include <QLabel>
+
+#include "OrbitClientData/ProcessData.h"
+#include "OrbitGgp/Instance.h"
+
+namespace orbit_qt {
+
+class TargetLabel : public QLabel {
+ public:
+  explicit TargetLabel(QWidget* parent = nullptr);
+  void SetFileTarget(const std::filesystem::path& path);
+  void SetStadiaTarget(const ProcessData& process, const orbit_ggp::Instance& instance);
+  void SetLocalTarget(const ProcessData& process);
+  void SetProcessDead();
+  void SetProcessAlive();
+  void SetConnectionDead(const QString& error_message);
+
+  static const QString kDefaultColor;
+  static const QString kGreenColor;
+  static const QString kOrangeColor;
+  static const QString kRedColor;
```

Comment:
Nit: These should be private, and I assume they are public because of the unit test? I'd avoid making something public just to be testable - it's good practice to test through the public API and ignore internals, so this indicates you're testing internals.

However - since this whole widget has purely visual purposes, this level of detail may be fine here... I just wanted to point this out, but I'm undecided.

Reply:
I made them an implementation detail and dont test for correct color anymore, only for change in color. ptal